src/transformers/modeling_utils.py

```diff
@@ -456,6 +475,11 @@ def init_weights(self):
         # Tie weights if needed
         self.tie_weights()
 
+        # Gradient accumulation if needed
+        if self.config.gradient_checkpointing:
+            for layer in self.get_layers():
+                layer.register_forward_pre_hook(self._gradient_accumulation_hook)
```

Comment:
Why not just replace the forward method here? I don't see what the hook adds since it seems like the only thing it does is:
```
layer.forward = functools.partial(torch.utils.checkpoint.checkpoint, layer.forward)
```

Reply:
You're correct, I was mistaken here.